docs/user-guide/node-selection/index.md

```diff
@@ -198,3 +198,215 @@ must be satisfied for the pod to schedule onto a node.
 
 For more information on inter-pod affinity/anti-affinity, see the design doc
 [here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/podaffinity.md).
+
+## Taints and tolerations (beta feature)
+
+### Specifying taints and tolerations
+
+Node affinity, described earlier, allows a pod to "be attracted to" a set of nodes
```

Comment:
It's a tough concept to explain. I think that contrasting with affinity is the most straightforward way to do it. Taints and tolerations are a newer-in-Kubernetes and more sophisticated concept than affinity, so I think people reading this section are likely to have some familiarity with affinity. Just to be on the safe side, I explain the salient properties of affinity in the sentence itself, so it should be self-contained even if the reader did not read the section on affinity.

I reworded the first paragraph a bit, and replaced the second paragraph with a modified version of the text you suggested.